### DISCUSSION TITLE
update, no rows, and select throws 404 exception

### DISCUSSION THREAD

**Discussion by sunilguptasg** [supabase/supabase#11266] [Questions]
an update(......).select(); throws this exception:

PostgrestException(message: [], code: 404, details: , hint: null)

if the update found no rows to update. 

This forces me to first run a select - verify if there are rows to update, and then update.

Is this a bug?

**Comment by GaryAustin1** [comment]
This was the way update operated for awhile, but new instances should return empty array without error.
https://github.com/supabase/postgrest-js/issues/294
If you have a free instance, pause/restore should get you the newer PostgREST (which is the Database API).

**Reply by steve-chavez** [reply]
Bug should be https://github.com/supabase/postgrest-js/issues/295
